### DISCUSSION TITLE
Decide what to do with the partial implementation of the orchard crate

### DISCUSSION THREAD

**Discussion by oxarbitrage** [ZcashFoundation/zebra#4903] [Ideas]
During Zcon3 there was some questions (Pablo from QEDIT) about the current use of the orchard crate in Zebra. Basically we use the upstream crate for some things, while our own implementation for others. The reason for this is that started our own implementation 
 back in the day when not everything we needed was available in the upstream crate.

The orchard crate is more mature now so probably the best thing to do will be to get rid of our own code in favor of the full dependency. However we need to check the details and make sure this is the way we want to go. If so, we can convert this discussion into a ticket.

**Comment by teor2345** [comment]
I'd be ok with this, we should think about it when we start work on integrating ZSAs into Zebra.

**Comment by mpguerra** [comment]
created #8155. Closing this discussion in favour of the issue
